kubernetes_state/check.py

```diff
@@ -299,40 +302,38 @@ def kube_pod_status_phase(self, message, **kwargs):
 
     def kube_pod_container_status_waiting_reason(self, message, **kwargs):
         metric_name = self.NAMESPACE + '.container.status_report.count.waiting'
-        whitelisted_reasons = {"ErrImagePull"}
-        reason = True
-        tags = []
         for metric in message.metric:
+            tags = []
+            skip_metric = False
             for label in metric.label:
                 if label.name == "reason":
-                    if label.value in whitelisted_reasons:
+                    if label.value in self.whitelisted_reasons['waiting']:
                         tags.append(self._format_tag(label.name, label.value))
                     else:
-                        reason = False
+                        report = True
```

Comment:
do you mean `skip_metric = True` ?

You can also start with `skip_metric = True` then put it to false when encountering a valid reason. That way you don't need the `else`